sockeye/layers.py

```diff
@@ -329,23 +304,17 @@ def hybrid_forward(self, F, queries, keys, values, lengths=None, bias=None):
         # (n, lq, lk)
         logits = F.batch_dot(lhs=queries, rhs=keys, transpose_b=True)
 
-        # TODO(fhieber): consider softmax with length argument once available.
-        # TODO(fhieber: Also see https://github.com/dmlc/gluon-nlp/pull/910
-        if lengths is not None:
-            # mask lk dimension
-            # (lk, n, lq)
-            logits = F.transpose(logits, axes=(2, 0, 1))
-            logits = F.SequenceMask(logits,
-                                    use_sequence_length=True,
-                                    sequence_length=lengths,
-                                    value=-C.LARGE_VALUES[self._dtype])
-            # (n, lq, lk)
-            logits = F.transpose(data=logits, axes=(1, 2, 0))
-
         if bias is not None:
+            assert lengths is None, "Cannot provide both bias and lengths"
             logits = F.broadcast_add(logits, bias)
 
-        probs = F.softmax(logits, axis=-1)
+        if lengths is not None:
+            assert bias is None, "Cannot provide both bias and lengths"
```

Comment:
Wouldn't it it be better to assert this only once and in the beginning of the method? e.g.
```python
assert not (bias and lengths), "Cannot provide both bias and lengths"
```

Reply:
Yeah, that would be a bit cleaner. Actually, there is no reason to have this assert really. Of course it doesn't really make sense to use both 'modes' usually, but why not allow it?